### PR TITLE
feat: add `dropRight` function

### LINKED ISSUE #134: feature `dropRight`
# Suggestion

Hello! I have used `fxjs` in my project.
And after apply typescript, i want to migrate from `fxjs` to `fxts`.
So, `fxts` API is not provide `dropRight` function yet.

## ⭐ Suggestion

please add `dropRight` function.

## 💻 Use Cases

```ts
pipe(
  [1, 2, 3, 4, 5],
  dropRight(1),
);
// [1, 2, 3, 4]
```

### PR DESCRIPTION
**Pull request by ppeeou**
Fixes #134

### CODE REVIEW

**Review by hg-pyun — APPROVED** [review]
LGTM 👍

File: src/Lazy/dropRight.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    yield arr[i];
+  }
+}
+
+function* sync<T>(length: number, iterable: Iterable<T>) {
+  const arr = toArray(iterable);
+  for (let i = 0; i < arr.length - length; i++) {
+    yield arr[i];
+  }
+}
```
Comment (shine1594) [inline src/Lazy/dropRight.ts:26]:
```suggestion
function* stringDropRight(length: number, str: string) {
  for (let i = 0; i < str.length - length; i++) {
    yield str[i];
  }
}

function* sync<T>(length: number, iterable: Iterable<T>) {
  const arr = isArray(iterable) ? iterable : toArray(iterable);
  for (let i = 0; i < arr.length - length; i++) {
    yield arr[i];
  }
}
```

Comment (ppeeou) [reply src/Lazy/dropRight.ts:26]: It seems consistent to proceed with `Array.from` for string in the same way as `reverse`...!

```ts
Array.from('👍🤔')
['👍', '🤔']
```

**Comment by Kuhave** [comment]
Don't forget to add a line of code to [website/function.json](https://github.com/marpple/FxTS/blob/main/website/function.json) file to see the description of the dropRight function on the API documentation website 😄

**Comment by ppeeou** [comment]
> Don't forget to add a line of code to [website/function.json](https://github.com/marpple/FxTS/blob/main/website/function.json) file to see the description of the dropRight function on the API documentation website 😄

Thanks 👍
If fxts release is decided after this PR
It will be uploaded to npm and added to website/function.json.

**Pull request merged**